ipatests/test_integration/test_sudo.py

```diff
@@ -756,3 +757,1299 @@ def test_domain_resolution_order(self):
+        ])
+
+    def test_006_sudorule_offline_caching_group_command(self):
+        master = self.master
+        kinit_admin(master)
+
+        master.run_command([
+            "ipa", "sudorule-add-allow-command", "--sudocmds=/bin/date",
+            self.SUDO_RULE
+        ])
+        master.run_command([
+            "ipa", "sudorule-remove-user", self.SUDO_RULE,
+            "--users", self.USER_1], raiseonerr=False
+        )
+        master.run_command([
+            "ipa", "group-add", "--desc=testgroup", self.GROUP
+        ])
+        master.run_command([
+            "ipa", "group-add-member", self.GROUP,
+            f"--users={self.USER_1}"
+        ])
+        master.run_command([
+            "ipa", "sudorule-add-user", self.SUDO_RULE,
+            f"--groups={self.GROUP}"], raiseonerr=False
+        )
+        clear_sssd_cache(self.client)
+
+        result = self.run_as_sudo_user(
+            "date", sudo_user="root", su_user=self.USER_1)
+
+        result = self.list_sudo_commands(self.USER_1)
+        assert "(root) /bin/date" in result.stdout_text
+
+        stop_ipa_server(master)
+        result = self.run_as_sudo_user(
+            "date", sudo_user="root", su_user=self.USER_1,
+            skip_kinit=True, skip_sssd_cache_clear=True)
+
+        result = self.list_sudo_commands(
+            self.USER_1, skip_kinit=True, skip_sssd_cache_clear=True
+        )
+        assert "(root) /bin/date" in result.stdout_text
+
+        start_ipa_server(master)
+        kinit_admin(master)
+        master.run_command([
+            "ipa", "sudorule-remove-allow-command",
+            "--sudocmds=/bin/date", self.SUDO_RULE
+        ])
+        master.run_command([
+            "ipa", "sudorule-remove-user", self.SUDO_RULE,
+            f"--groups={self.GROUP}"], raiseonerr=False
+        )
+        master.run_command([
+            "ipa", "sudorule-add-user", self.SUDO_RULE,
+            f"--users={self.USER_1}"], raiseonerr=False
+        )
+        master.run_command([
+            "ipa", "group-remove-member", self.GROUP,
+            f"--users={self.USER_1}"
+        ])
+        master.run_command([
+            "ipa", "group-del", self.GROUP])
+
+    def test_007_sudorule_offline_caching_option_command(self):
+        master = self.master
+        kinit_admin(master)
+
+        master.run_command([
+            "ipa", "sudorule-add-allow-command", "--sudocmds=/bin/date",
+            self.SUDO_RULE
+        ])
+        master.run_command([
+            'ipa', 'sudorule-add-option', self.SUDO_RULE,
+            '--sudooption', "!authenticate"
+        ])
+        clear_sssd_cache(self.client)
+        result = self.run_as_sudo_user(
+            "date", sudo_user="root", su_user=self.USER_1,
+            skip_password=True)
+
+        result = self.list_sudo_commands(self.USER_1)
+        assert "(root) NOPASSWD: /bin/date" in result.stdout_text
+
+        stop_ipa_server(master)
+        result = self.run_as_sudo_user(
```

Comment:
The test doesn't check the result, either use the value in an assertion or replace with self.run_as_sudo_user(